packages/shared/src/server/definitions.ts

```diff
@@ -172,3 +172,130 @@ export const convertScoreReadToInsert = (
+ * column names, not the Prisma mapped names.
+ */
+export const convertPostgresObservationToInsert = (
+  observation: Record<string, any>,
+): ObservationRecordInsertType => {
+  return {
+    id: observation.id,
+    trace_id: observation.trace_id,
+    project_id: observation.project_id,
+    type: observation.type,
+    parent_observation_id: observation.parent_observation_id,
+    start_time: observation.start_time?.getTime(),
+    end_time: observation.end_time?.getTime(),
+    name: observation.name,
+    metadata:
+      typeof observation.metadata === "string"
+        ? { metadata: observation.metadata }
+        : Array.isArray(observation.metadata)
+          ? { metadata: observation.metadata }
+          : observation.metadata,
+    level: observation.level,
+    status_message: observation.status_message,
+    version: observation.version,
+    input: observation.input ? JSON.stringify(observation.input) : null,
+    output: observation.output ? JSON.stringify(observation.output) : null,
+    provided_model_name: observation.model,
+    internal_model_id: observation.internal_model_id,
+    model_parameters: observation.model_parameters,
+    provided_usage_details: {},
+    usage_details: {
+      input: observation.prompt_tokens,
+      output: observation.completion_tokens,
+      total: observation.total_tokens,
+    },
+    provided_cost_details: {
+      input: observation.input_cost,
+      output: observation.output_cost,
+      total: observation.total_cost,
+    },
+    cost_details: {
+      input: observation.calculated_input_cost,
+      output: observation.calculated_output_cost,
+      total: observation.calculated_total_cost,
+    },
+    total_cost: observation.calculated_total_cost,
+    completion_start_time: observation.completion_start_time?.getTime(),
+    prompt_id: observation.prompt_id,
+    prompt_name: observation.prompt_name,
+    prompt_version: observation.prompt_version,
+    created_at: observation.created_at?.getTime(),
+    updated_at: observation.updated_at?.getTime(),
+    event_ts: observation.start_time?.getTime(),
+    is_deleted: 0,
+  };
+};
+
+/**
+ * Expects a single record from a `select * from scores` query. Must be a raw query to keep original
+ * column names, not the Prisma mapped names.
+ */
+export const convertPostgresScoreToInsert = (
+  score: Record<string, any>,
+): ScoreRecordInsertType => {
+  return {
+    id: score.id,
+    timestamp: score.timestamp?.getTime(),
+    project_id: score.project_id,
+    trace_id: score.trace_id,
+    observation_id: score.observation_id,
+    name: score.name,
+    value: score.value,
+    source: score.source,
+    comment: score.comment,
+    author_user_id: score.author_user_id,
+    config_id: score.config_id,
+    data_type: score.data_type,
+    string_value: score.string_value,
+    created_at: score.created_at?.getTime(),
+    updated_at: score.updated_at?.getTime(),
+    event_ts: score.timestamp?.getTime(),
+    is_deleted: 0,
```

Comment:
queue_id is missing here, i think

Reply:
Added in https://github.com/langfuse/langfuse/pull/3978/commits/42ebd90c136e753633f636956ab4a71d90eb0f21